src/python/pants/backend/native/subsystems/native_toolchain.py

```diff
@@ -329,31 +329,31 @@ class ToolchainVariantRequest:
 
 @rule
 def select_c_toolchain(toolchain_variant_request: ToolchainVariantRequest) -> CToolchain:
-  native_toolchain = toolchain_variant_request.toolchain
-  # TODO(#5933): make an enum exhaustiveness checking method that works with `yield Get(...)`!
-  use_gcc = toolchain_variant_request.variant.resolve_for_enum_variant({
-    'gnu': True,
-    'llvm': False,
-  })
+  use_gcc = {
```

Comment:
In this particular case, I think pattern matching is a tad overkill:
`use_gcc = (toolchain_variant_request.variant == ToolChainVariant.gnu)` reads better to me.

(If you think of it, if you added a third variant, the line above would still be fine whereas the pattern matching one would need modification)